app/screens/Recognition/RecognitionDialog/components/Dialog/Dialog.tsx

```diff
@@ -0,0 +1,39 @@
+import _ from 'lodash';
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { Image, TouchableOpacity, View } from 'react-native';
+
+import { IDialogProps } from './Dialog.types';
+import { styles } from './Dialog.styles';
+import { ExtendedText } from '../../../../../components';
+import { IMAGES } from '../../../../../assets';
+
+export const Dialog: React.FC<IDialogProps> = ({
+  data,
+  onNextPress,
+  currentIndex,
+}) => {
+  const { t } = useTranslation();
+
+  return (
+    <View style={styles.container}>
+      <View style={styles.dialogContainer}>
+        {_.map(
+          data,
+          (item, index) =>
+            index === currentIndex && (
+              <ExtendedText style={styles.title} key={item.textKey}>
+                {t(item.textKey)}
+              </ExtendedText>
+            ),
+        )}
```

Comment:
so the result will be
```
[false, false, false, JSX, false, false]
```

please, refactor to avoid array if we need ONLY one element to render -> you can use `_.find` before JSX and then render it without map